Performance E2E (one-shot deploy) deploy intermittently fails preparing relay network JSON

## Summary

Performance E2E tests frequently fail during the one-shot deploy `before all` hook while preparing relay Helm chart values.

## Failing jobs

- https://github.com/hiero-ledger/solo/actions/runs/28473108183/job/84390681223
- https://github.com/hiero-ledger/solo/actions/runs/28473112046/job/84390694331

## Error

```text
2 failing

1) E2E Performance Tests Suite
     E2E Performance Tests
       "before all" hook for "TokenTransferLoadTest":
   OneShotDeployFailedSoloError: Deploy failed: Cannot read properties of undefined (reading 'haProxyClusterIp'). Rollback skipped (--no-rollback).\n    at DefaultOneShotCommand.performRollback (/home/runner/_work/solo/solo/src/commands/one-shot/default-one-shot.ts:205:13)\n    at DefaultOneShotCommand.deployInternal (/home/runner/_work/solo/solo/src/commands/one-shot/default-one-shot.ts:276:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:103:5)\n    at async Object.handler (/home/runner/_work/solo/solo/src/core/command-path-builders/command-builder.ts:141:47)\n   Caused by: TypeError: Cannot read properties of undefined (reading 'haProxyClusterIp')\n    at RelayCommand.prepareNetworkJsonString (/home/runner/_work/solo/solo/src/commands/relay.ts:367:76)\n    at process.processTicksAndRejections (node:internal/process/task_queues:103:5)\n    at async RelayCommand.prepareHelmChartValuesForRelay (/home/runner/_work/solo/solo/src/commands/relay.ts:326:39)\n    at async Task.task (/home/runner/_work/solo/solo/src/commands/relay.ts:419:39)\n    at async Task.run (/home/runner/_work/solo/solo/node_modules/listr2/dist/index.mjs:1908:5)\n```\n\n## Initial suspicion\n\n`RelayCommand.prepareNetworkJsonString()` assumes every node ID in the relay deployment has a corresponding node service mapping entry with `haProxyClusterIp`. In this failure path, at least one lookup returns `undefined`, causing the deploy to fail with a generic TypeError instead of waiting/retrying or reporting which node/service mapping is missing.\n\n## Debug notes\n\nNeed inspect relay service mapping construction and the one-shot/performance deploy path to determine whether this is a race while service endpoints are still being discovered or an invalid node alias selection for the relay values.

--- a/.github/workflows/script/launch_network.sh
+++ b/.github/workflows/script/launch_network.sh
@@ -625,7 +625,8 @@ BLOCK_NODE_VERSION="${PREV_BLOCK_VERSION#v}" \
   solo one-shot falcon deploy \
   --num-consensus-nodes 2 \
   --consensus-node-version "${FROM_CONSENSUS_NODE_VERSION}" \
-  --values-file "${TEMP_ONE_SHOT_VALUES_FILE}"
+  --values-file "${TEMP_ONE_SHOT_VALUES_FILE}" \
+  --no-parallel-deploy
 
 echo "::endgroup::"
 

--- a/src/commands/one-shot/orchestrator/deploy/default-one-shot-deploy-orchestrator.ts
+++ b/src/commands/one-shot/orchestrator/deploy/default-one-shot-deploy-orchestrator.ts
@@ -731,10 +731,12 @@ export class DefaultOneShotDeployOrchestrator implements OneShotDeployOrchestrat
                 this.taskList,
                 (): boolean => !getConfig().deployRelay && !getConfig().minimalSetup,
               ),
-          }).withWaitCondition(
-            SoloEventType.MirrorNodeDeployed,
-            Duration.ofMinutes(constants.MIRROR_NODE_DEPLOYED_EVENT_TIMEOUT_MINUTES),
-          ),
+          })
+            .withWaitCondition(
+              SoloEventType.MirrorNodeDeployed,
+              Duration.ofMinutes(constants.MIRROR_NODE_DEPLOYED_EVENT_TIMEOUT_MINUTES),
+            )
+            .withWaitCondition(SoloEventType.NodesStarted, Duration.ofMinutes(10)),
         ],
         (getConfig: () => OneShotSingleDeployConfigClass): ExecutionMode =>
           getConfig().parallelDeploy

--- a/src/commands/one-shot/orchestrator/deploy/default-one-shot-deploy-orchestrator.ts
+++ b/src/commands/one-shot/orchestrator/deploy/default-one-shot-deploy-orchestrator.ts
@@ -708,10 +708,12 @@ export class DefaultOneShotDeployOrchestrator implements OneShotDeployOrchestrat
                 this.taskList,
                 (): boolean => !getConfig().deployMirrorNode || !getConfig().pinger,
               ),
-          }).withWaitCondition(
-            SoloEventType.MirrorNodeDeployed,
-            Duration.ofMinutes(constants.MIRROR_NODE_DEPLOYED_EVENT_TIMEOUT_MINUTES),
-          ),
+          })
+            .withWaitCondition(
+              SoloEventType.MirrorNodeDeployed,
+              Duration.ofMinutes(constants.MIRROR_NODE_DEPLOYED_EVENT_TIMEOUT_MINUTES),
+            )
+            .withWaitCondition(SoloEventType.NodesStarted, Duration.ofMinutes(10)),
           new OrchestratorPipelinePhase('Deploy explorer', {
             asListrTask: (getConfig: () => OneShotSingleDeployConfigClass): SoloListrTask<OneShotSingleDeployContext> =>
               invokeSoloCommand(

